pkg/controller/apicast/apicast_controller.go

```diff
@@ -189,6 +188,34 @@ func (r *ReconcileAPIcast) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileAPIcast) updateStatus(instance *appsv1alpha1.APIcast, reconciler *APIcastLogicReconciler) (reconcile.Result, error) {
```

Comment:
Where is this function called?

Reply:
Fixed.

Output when deploying APIcast with the operator:

```
msoriano@localhost:~/go/src/github.com/3scale/apicast-operator/example-manifests (add-image-to-status)$ oc get apicast example-apicast -o yaml
apiVersion: apps.3scale.net/v1alpha1
kind: APIcast
metadata:
  annotations:
    apicast.apps.3scale.net/operator-version: 0.1.0
  creationTimestamp: 2020-01-22T15:39:38Z
  generation: 2
  name: example-apicast
  namespace: msoriano-test
  resourceVersion: "178001"
  selfLink: /apis/apps.3scale.net/v1alpha1/namespaces/msoriano-test/apicasts/example-apicast
  uid: 663f96ed-3d2d-11ea-b510-0800271e62c0
spec:
  embeddedConfigurationSecretRef:
    name: apicast-gateway-configuration
  replicas: 1
status:
  image: quay.io/3scale/apicast:nightly
```

If I overwrite the image field in spec the image field in status gets updated accordingly:

```
msoriano@localhost:~/go/src/github.com/3scale/apicast-operator/example-manifests (add-image-to-status)$ oc get apicast example-apicast -o yaml
apiVersion: apps.3scale.net/v1alpha1
kind: APIcast
metadata:
  annotations:
    apicast.apps.3scale.net/operator-version: 0.1.0
  creationTimestamp: 2020-01-22T15:39:38Z
  generation: 3
  name: example-apicast
  namespace: msoriano-test
  resourceVersion: "178922"
  selfLink: /apis/apps.3scale.net/v1alpha1/namespaces/msoriano-test/apicasts/example-apicast
  uid: 663f96ed-3d2d-11ea-b510-0800271e62c0
spec:
  embeddedConfigurationSecretRef:
    name: apicast-gateway-configuration
  image: quay.io/3scale/apicast:v3.7.0
  replicas: 1
status:
  image: quay.io/3scale/apicast:v3.7.0
```